src/utils/privateRoute.js

```diff
@@ -1,9 +1,8 @@
 import React from "react";
-import { useSelector } from "react-redux";
 import { Navigate } from "react-router-dom";
-useSelector(state=>state)
 
 // eslint-disable-next-line react/prop-types
-export function PrivateRoute() {
-  return localStorage.getItem("token") ? <Navigate to='/dashboard' /> : <Navigate to='/' /> 
+export function PrivateRoute({ children }) {
+  const auth = localStorage.getItem("token");
```

Comment:
@Aqeeb12 use reducer here instead of localstroage , initiate reducer with localstorage flag so you have right status on reducer when reloading